services/apps/alcs/src/portal/notification-submission/notification-submission.controller.spec.ts

```diff
@@ -0,0 +1,264 @@
+    ]);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  it('should call out to service when fetching notice of intents', async () => {
+    mockNoiSubmissionService.getAllByUser.mockResolvedValue([]);
+
+    const submissions = await controller.getSubmissions({
+      user: {
+        entity: new User(),
+      },
+    });
+
+    expect(submissions).toBeDefined();
+    expect(mockNoiSubmissionService.getAllByUser).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call out to service when fetching a notice of intent', async () => {
+    mockNoiSubmissionService.mapToDetailedDTO.mockResolvedValue(
+      {} as NotificationSubmissionDetailedDto,
+    );
+
+    const noticeOfIntent = controller.getSubmission(
+      {
+        user: {
+          entity: new User(),
+        },
+      },
+      '',
+    );
+
+    expect(noticeOfIntent).toBeDefined();
+    expect(mockNoiSubmissionService.getByUuid).toHaveBeenCalledTimes(1);
+  });
+
+  it('should fetch notice of intent by bceid if user has same guid as a local government', async () => {
+    mockNoiSubmissionService.mapToDetailedDTO.mockResolvedValue(
+      {} as NotificationSubmissionDetailedDto,
+    );
+    mockNoiSubmissionService.getByUuid.mockResolvedValue(
+      new NotificationSubmission({
+        localGovernmentUuid: '',
+      }),
+    );
+
+    const noiSubmission = controller.getSubmission(
+      {
+        user: {
+          entity: new User({
+            bceidBusinessGuid: 'guid',
+          }),
+        },
+      },
+      '',
+    );
+
+    expect(noiSubmission).toBeDefined();
+    expect(mockNoiSubmissionService.getByUuid).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call out to service when creating an notice of intent', async () => {
+    mockNoiSubmissionService.create.mockResolvedValue('');
+    mockNoiSubmissionService.mapToDTOs.mockResolvedValue([
+      {} as NotificationSubmissionDto,
+    ]);
+
+    const noiSubmission = await controller.create({
+      user: {
+        entity: new User(),
+      },
+    });
+
+    expect(noiSubmission).toBeDefined();
+    expect(mockNoiSubmissionService.create).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call out to service for update and map', async () => {
+    mockNoiSubmissionService.mapToDetailedDTO.mockResolvedValue(
+      {} as NotificationSubmissionDetailedDto,
+    );
+    mockNoiSubmissionService.getByUuid.mockResolvedValue(
+      new NotificationSubmission({}),
+    );
+
+    await controller.update(
+      'file-id',
+      {
+        localGovernmentUuid,
+        applicant,
+      },
+      {
+        user: {
+          entity: new User({
+            clientRoles: [],
+          }),
+        },
+      },
+    );
+
+    expect(mockNoiSubmissionService.update).toHaveBeenCalledTimes(1);
+    expect(mockNoiSubmissionService.mapToDetailedDTO).toHaveBeenCalledTimes(1);
+  });
+
+  // it('should throw an exception when trying to update a not in progress noi', async () => {
```

Comment:
is this still relevant?

Reply:
yes, will be uncommented when we add status